AdyenUI/Styles/AdyenTextFieldStyle.swift

```diff
@@ -0,0 +1,150 @@
+        self.backgroundColor = backgroundColor
+        self.textColor = textColor
+        self.activeColor = activeColor
+        self.errorColor = errorColor
+        self.cornerRadius = cornerRadius
+        self.borderColor = borderColor
+        self.borderWidth = borderWidth
+    }
+
+    public init() {
+        self.backgroundColor = AdyenColorScheme.default.background
+        self.textColor = AdyenColorScheme.default.text
+        self.activeColor = AdyenColorScheme.default.highlight
+        self.errorColor = AdyenColorScheme.default.destructive
+        self.cornerRadius = CornerRounding.fixed(AdyenUIConstants.defaultCornerRadius)
+        self.borderColor = AdyenColorScheme.default.outline
+        self.borderWidth = AdyenUIConstants.defaultBorderWidth
+    }
+}
+
+// MARK: - Method Chaining for AdyenTextFieldStyle Properties
+
+// This extension adds the method chaining to the AdyenTextFieldStyle struct.
+extension AdyenTextFieldStyle {
+    
+    /// Returns a new style with the specified background color.
+    /// - Parameter backgroundColor: The color to set.
+    /// - Returns: A new `AdyenTextFieldStyle` instance.
+    public func with(backgroundColor: UIColor) -> AdyenTextFieldStyle {
+        var newStyle = self
+        newStyle.backgroundColor = backgroundColor
+        return newStyle
+    }
+    
+    /// Returns a new style with the specified text color.
+    /// - Parameter textColor: The color to set.
+    /// - Returns: A new `AdyenTextFieldStyle` instance.
+    public func with(textColor: UIColor) -> AdyenTextFieldStyle {
+        var newStyle = self
+        newStyle.textColor = textColor
+        return newStyle
+    }
+    
+    /// Returns a new style with the specified active color.
+    /// - Parameter activeColor: The color to set.
+    /// - Returns: A new `AdyenTextFieldStyle` instance.
+    public func with(activeColor: UIColor) -> AdyenTextFieldStyle {
+        var newStyle = self
+        newStyle.activeColor = activeColor
+        return newStyle
+    }
+    
+    /// Returns a new style with the specified error color.
+    /// - Parameter errorColor: The color to set.
+    /// - Returns: A new `AdyenTextFieldStyle` instance.
+    public func with(errorColor: UIColor) -> AdyenTextFieldStyle {
+        var newStyle = self
+        newStyle.errorColor = errorColor
+        return newStyle
+    }
+    
+    /// Returns a new style with the specified corner radius.
+    /// - Parameter cornerRadius: The corner radius to set.
+    /// - Returns: A new `AdyenTextFieldStyle` instance.
+    public func with(cornerRadius: CornerRounding) -> AdyenTextFieldStyle {
+        var newStyle = self
+        newStyle.cornerRadius = cornerRadius
+        return newStyle
+    }
+    
+    /// Returns a new style with the specified border color.
+    /// - Parameter borderColor: The color to set.
+    /// - Returns: A new `AdyenTextFieldStyle` instance.
+    public func with(borderColor: UIColor) -> AdyenTextFieldStyle {
```

Comment:
I find it a bit confusing from a merchant perspective, it's not clear if this `borderColor` would be always applied including active and non-active states, as we already have a dedicated semantic color for `activeColor`, which can override default `borderColor`.